File: Solarness.Model/Team.cs
using System;
using System.Collections.Generic;

namespace Solarness.Model;

public partial class Team
{
    public int TeamId { get; set; }

    public string TeamName { get; set; } = null!;

    public string? Description { get; set; }

    public DateTime? CreationDate { get; set; }

    public int? UserId { get; set; }

    public virtual User? User { get; set; }
}

File: UI/solarness_desktop_app/lib/models/Team/team.dart
import 'package:json_annotation/json_annotation.dart';

part 'team.g.dart';

@JsonSerializable()
class Team {
  int? teamId;
  String? teamName;
  String? description;
  DateTime? creationDate;

  Team(this.teamId, this.teamName, this.description, this.creationDate);

  factory Team.fromJson(Map<String, dynamic> json) =>
      _$TeamFromJson(json);

  Map<String, dynamic> toJson() => _$TeamToJson(this);
}
